docs/logs/complex-objects/Program.cs

```diff
@@ -23,10 +26,12 @@
     CompanyName = "Contoso Fresh Vegetables, Inc.",
 };
 
-logger.FoodRecallNotice(foodRecallNotice);
-
-// Dispose logger factory before the application ends.
-// This will flush the remaining logs and shutdown the logging pipeline.
+logger.FoodRecallNotice(foodRecallNotice);
+
+// This will flush the remaining logs.
+loggerProvider.ForceFlush();

```

Comment:
wondering how this works. the `loggerProvider` seems to have no connection to the LoggerFactory?

Reply:
Agree - was going to ask some comments in here actually and you got to this fast :).

Reply:
Generally:

- Unclear to me if this needs to be before LoggerFactory.Create and if they interact at all.  Unclear how to get a LoggerProvider outside of traditional DI.  We could convert all of these examples to look like this:
-- https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#create-an-iloggerfactory-with-di 
- Unintituive that this `LoggerProvider` is not an `ILoggerProvider` (but this can't be helped here)
- In general, I think the `Sdk.Create*()` APIs are going to add confusion with the existing .Net blogs posts of how to use ILogger, ILoggerProvider, and ILoggerFactory.
-

Reply:
`Sdk.CreateLoggerProviderBuilder()` is an experimental API, not available in stable release. It's best to *not* show its usage here before stabilizing it.

Sorry I don't know if there is any way to do force_flush() on LoggerProvider unless the above happens.

> Sdk.Create*() APIs are going to add confusion with the existing .Net blogs posts of how to use ILogger, ILoggerProvider, and ILoggerFactory.

Not sure what you meant by that. `ILogger` itself has 2 ways of doing things - one for DI, and one for non-DI. Sdk.Create* in OTel is OTel's non-di way. For DI, extensions methods on `IServiceCollection` should be preferred.

(Irrespective of the above, fully agree that the docs/examples should show how to obtain a provider so that users can explicitly call shutdown() (or flush()) on it themselves, rather than relying on automatic shutdown upon dispose, which uses 5 sec max to perform shutdown. It's surprising that this was never reported as an issue before. Maybe the 5 sec was liberal enough for majority users?)

Reply:
Here is how you do it using the current stable API:

```csharp
var sdk = OpenTelemetrySdk.Create(builder => builder
    .WithLogging(logging => logging.AddConsoleExporter()));

var logger = sdk.GetLoggerFactory().CreateLogger<Program>();

var foodRecallNotice = new FoodRecallNotice
{
    BrandName = "Contoso",
    ProductDescription = "Salads",
    ProductType = "Food & Beverages",
    RecallReasonDescription = "due to a possible health risk from Listeria monocytogenes",
    CompanyName = "Contoso Fresh Vegetables, Inc.",
};

logger.FoodRecallNotice(foodRecallNotice);

sdk.LoggerProvider.ForceFlush();

// Dispose SDK before the application ends.
sdk.Dispose();
```

Reply:
Thanks @CodeBlanch - let me try this out.

Reply:
@CodeBlanch With this new API, is it possible for users to add other non-otel providers like `ConsoleProvider` etc.? It the builder the ILoggerBuilder, so one can add other providers like they normally do?

Reply:
Gave it a whirl, added a console di sample as well.

Reply:
> With this new API, is it possible for users to add other non-otel providers like `ConsoleProvider` etc.? It the builder the ILoggerBuilder, so one can add other providers like they normally do?

@cijothomas Yes it is possible, but a bit awkward:

```csharp
var sdk = OpenTelemetrySdk.Create(builder =>
{
    // Configure OTel logging pipeline
    builder.WithLogging(
        logging => logging.AddOtlpExporter());

    // Configure ILogger pipeline to add additional non-otel providers
    builder.Services.AddLogging(
        logging => logging.AddConsole());
});
```

It could be made easier to do with an extension (or whatever) but it isn't there today.

Reply:
I see. In that case, would it make sense to keep the existing examples as-is, so we don't accidentally cause users to create a *new* Logger pipeline, detached from the one with their existing LoggerProviders.

Reply:
FWIW - I would vote for replacing of all `LoggerFactory.Create()` examples given that if/when you have to use something from the sdk, you will have to change off of it - and these samples are around using said otel sdk?